src/utils.ts

```diff
@@ -0,0 +1,27 @@
+export type Color = {
+    name: string;
+    hex: string;
+    comp?: Color[];
+};
+
+/*
+ * Filter array of colors to ensure no duplicate values are present.
+ * @param colors - The Color array
+ * @returns The Color array less any duplicate values
+ */
+export const filterColors = (colors: Color[] | undefined) => {
```

Comment:
This can return undefined, which can be a headache to handle elsewhere in the app. You could consider returning an empty array if passed in undefined.